src/i18n.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { i18n } from '@lingui/core'
+import { I18nProvider } from '@lingui/react'
+import { detect, fromUrl } from '@lingui/detect-locale'
+import { ReactNode, useEffect } from 'react'
+
+export const locales = ['en', 'it']
+export const defaultLocale = 'en'
+
+locales.map((locale) => i18n.loadLocaleData(locale, { plurals: () => null }))
+
+const getDetectedLocale = () => {
+  const detected =
+    detect(
+      fromUrl('lang'), // helps local development
+      defaultLocale
+    ) ?? defaultLocale
+  return locales.includes(detected) ? detected : defaultLocale
+}
+
+export async function dynamicActivate(locale: string) {
+  const { messages } = await import(`@lingui/loader!./locales/${locale}/messages.po`)
```

Comment:
isn't it going to be included in the build like this? also interpolating the locale makes me think webpack would not know to include the file. how would this be excluded from the production build?

Reply:
lingui.config.js has a default locale which I believe will always be included in the bundle.

Non-default locales will be downloaded on the fly.